packages/server-wallet/src/models/objective.ts

```diff
@@ -131,21 +131,28 @@ export class ObjectiveModel extends Model {
     const id: string = objectiveId(objectiveToBeStored);
 
     return tx.transaction(async trx => {
-      const model = await ObjectiveModel.query(trx).insert({
-        objectiveId: id,
-        status: objectiveToBeStored.status,
-        type: objectiveToBeStored.type,
-        data: objectiveToBeStored.data,
-        createdAt: new Date().toISOString(),
-        progressLastMadeAt: new Date().toISOString(),
-      });
+      const model = await ObjectiveModel.query(trx)
+        .insert({
+          objectiveId: id,
+          status: objectiveToBeStored.status,
+          type: objectiveToBeStored.type,
+          data: objectiveToBeStored.data,
+          createdAt: new Date().toISOString(),
+          progressLastMadeAt: new Date().toISOString(),
+        })
+        .onConflict('objectiveId')
+        .ignore(); // this makes it an upsert
```

Comment:
Should this return `*` so that we always get the latest row?

Reply:
Yes. We also need to decide what to do if say `OpenChannel-0x123` exists with status 'pending', and we try to insert `OpenChannel-0x123` with status 'approved'. This might be an argument for doing `onConflict('objectiveId').merge({status: objectiveToBeStored.status})`. Either way we need some unit tests for this so we can assert on the desired behaviour.